### PR TITLE
feat: added log_analytics_workspace in foundation/core and lz/aks

### PR DESCRIPTION
**Pull request by delirius325**
This adds a `Log Analytics Workspace` in the management resource group.

** Before merging, can we test the AKS part of it? I don't have sufficient permissions to do so **

### CODE REVIEW

File: foundation/core/monitoring.tf [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  name     = "rg-${var.prefix}-management"
+  location = var.location
+}
+
+resource "azurerm_log_analytics_workspace" "log_analytics_workspace" {
+  provider            = azurerm.management
+  name                = "log-${var.prefix}-management"
+  location            = azurerm_resource_group.caf_management.location
+  resource_group_name = azurerm_resource_group.caf_management.name
+  sku                 = "PerGB2018"
```
Comment (delirius325) [inline foundation/core/monitoring.tf:12]: Do we want to have a variable for this?

Comment (christopherswarren) [reply foundation/core/monitoring.tf:12]: I would say yes. Good catch

File: modules/azure/aks/main.tf [hunk 2]
```diff
@@ -57,7 +57,8 @@ resource "azurerm_kubernetes_cluster" "aks" {
     }
 
     oms_agent {
-      enabled = false
+      enabled                    = false
```
Comment (djtjwillia) [inline modules/azure/aks/main.tf:60]: wouldn't this agent need to be enabled to have it output logs?

**Review by christopherswarren — APPROVED** [review]
👍

**Pull request merged**